lib/Doctrine/ORM/Proxy/ProxyFactory.php

```diff
@@ -180,4 +291,76 @@ private function createCloner(ClassMetadata $classMetadata, EntityPersister $ent
             }
         };
     }
+
+    private function generateUseLazyGhostTrait(ClassMetadata $class): string
+    {
+        $code = ProxyHelper::generateLazyGhost($class->getReflectionClass());
+        $code = substr($code, 7 + (int) strpos($code, "\n{"));
+        $code = substr($code, 0, (int) strpos($code, "\n}"));
+        $code = str_replace('LazyGhostTrait;', str_replace("\n    ", "\n", 'LazyGhostTrait {
+            initializeLazyObject as __load;
+            setLazyObjectAsInitialized as public __setInitialized;
+            isLazyObjectInitialized as private;
+            createLazyGhost as private;
+            resetLazyObject as private;
+            __clone as private __doClone;
+        }'), $code);
+
+        return $code;
+    }
+
+    private function generateSkippedProperties(ClassMetadata $class): string
+    {
+        $skippedProperties = ['__isCloning' => true];
+        $identifiers       = array_flip($class->getIdentifierFieldNames());
+
+        foreach ($class->getReflectionClass()->getProperties() as $property) {
+            $name = $property->getName();
+
+            if ($property->isStatic() || (($class->hasField($name) || $class->hasAssociation($name)) && ! isset($identifiers[$name]))) {
+                continue;
+            }
+
+            $prefix = $property->isPrivate() ? "\0" . $property->getDeclaringClass()->getName() . "\0" : ($property->isProtected() ? "\0*\0" : '');
+
+            $skippedProperties[$prefix . $name] = true;
+        }
+
+        uksort($skippedProperties, 'strnatcmp');
+
+        $code = VarExporter::export($skippedProperties);
+        $code = str_replace(VarExporter::export($class->getName()), 'parent::class', $code);
+        $code = str_replace("\n", "\n        ", $code);
+
+        return $code;
+    }
+
+    private function generateSerializeImpl(ClassMetadata $class): string
+    {
+        $reflector  = $class->getReflectionClass();
+        $properties = $reflector->hasMethod('__serialize') ? 'parent::__serialize()' : '(array) $this';
+
+        $code = '$properties = ' . $properties . ';
+        unset($properties["\0" . self::class . "\0lazyObjectId"]);
```

Comment:
shouldn't this also using `__isCloning` ?

Reply:
And shouldn't this `unset` be done only for the array cast and not for `parent::__serialize()` ?

Reply:
updated, this won't change much in practice but that's still better to unset indeed.
About your second question: no, the property should never be serialized, even if the parent returns it.
FYI, it is legit to implement __serialize without implementing __unserialize. When doing to, __serialize shall return an array structure modeled after `(array) $this`. This is when the property can leak and should be unset.